src/nvim/regexp.c

```diff
@@ -1389,6 +1389,10 @@ int vim_regcomp_had_eol(void)
   return had_eol;
 }
 
+// variables for parsing reginput
+static int at_start;       // True when on the first character
+static int prev_at_start;  // True when on the second character
```

Comment:
original patch moved (deleted) the declarations from line 2691, I included that in the merge.